### PR TITLE
Create dates screen on course creator form

### PR DESCRIPTION
**Pull request by Cressence**
## Before the Pull Request ##
* Test out your changes manually to confirm that it all works and search for bugs.
* Run the Javascript tests locally (`yarn test`) and confirm that they are passing.
* Run the specs for any Ruby files you changed, along with the feature specs that are related to your changes. Optionally, you can run the entire test suite locally (`rake spec`).

## Opening the Pull Request ##

* Describe the changes included in the PR.
* Note the issue that the PR addresses. Include `Fixes #12345` if it will completely address the issue.
* If there are UI elements to the change, include a screenshot or animation to illustrate it.
* If the PR is not complete but you want feedback on it or you just want to trigger a build, include `[WIP]` in the title to indicate work-in-progress, and add comments about anything you're stuck on. Edit the title when it's ready.

## After the PR ##
* Check the continuous integration build, which usually takes about 20 minutes.
    * Fix any failures.
    * Add a comment to the PR if you are stuck. If the build is failing and you don't discuss it with anyone, we will assume that you've seen the failing specs and are working to fix them.
    * Occasionally, specs unrelated to the PR will fail. If you think that's the case, you can ping someone to restart the build, or close and reopen the PR to trigger a new build.
* If the build is passing, we will review it as soon as we can.
* If you are abandoning a PR, please close it to help us keep track of which PRs are still active.

### CODE REVIEW

File: spec/features/course_creation_spec.rb [hunk 1]
```diff
@@ -130,10 +132,20 @@ def go_through_researchwrite_wizard
...
+      sleep 3
+
       find('#course_title').set('My awesome new course - Foo 101')
 
-      # If we click before filling out all require fields, only the invalid
-      # fields get restyled to indicate the problem.
+      click_button 'Create my Course!'
+
+      If we click before filling out all require fields, only the invalid
+      fields get restyled to indicate the problem.
```
Comment (ragesoss) [inline spec/features/course_creation_spec.rb]: These two lines got accidentally uncommented.

Comment (Cressence) [reply spec/features/course_creation_spec.rb]: Oh okay

**Comment by ragesoss** [comment]
Sorry for not getting to this last week; this is working now, yay!

A few things before we can merge this:
* The timeline start/end dates should not be included at all for the ClassroomProgramCourse case. This is because the first step of the Wizard is where that info is handled.
* The 'Your role in the course' field has been changed in the latest version on master, and that change needs to be incorporated here. (It's been turned into a hybrid select field with some common values available as a pulldown.)

**Comment by Cressence** [comment]
> Sorry for not getting to this last week; this is working now, yay!
> 
> A few things before we can merge this:
> 
> * The timeline start/end dates should not be included at all for the ClassroomProgramCourse case. This is because the first step of the Wizard is where that info is handled.
> * The 'Your role in the course' field has been changed in the latest version on master, and that change needs to be incorporated here. (It's been turned into a hybrid select field with some common values available as a pulldown.)

I pulled from master but didn't see the select field

**Comment by ragesoss** [comment]
There are a couple more UI problems I found testing the latest version.

For the layout, the location of the role field should remain the same, below the Description field. Also, the Next button should be aligned with the Back button.

The validation works as expected for title, term, institution and expected students fields, but you can proceed to the dates step without filling out the Description field... and then you get the validation error at the last step if you proceed without filling out the description.

There should also be a Back button on the dates step.

**Pull request merged**